frontend/lib/services/persistency_service.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:async';
+
+import 'package:frontend/models/event.dart';
+import 'package:frontend/models/sailing_club.dart';
+import 'package:frontend/models/boat.dart';
+
+// TODO: Should we have a percistency Interface that is segregated from the authentication interface?
+
+abstract class PersistencyService {
+//  final RegattaStore _store;
+
+//  PersistencyService(this._store);
+
+  Future signIn();
+  void signOut();
+
+  // These private methods are called by implementing classes
+  // to propagate changes in the DB to the frontend.
+/*
+  void newEvent(Event newEvent) {
```

Comment:
Currently those are commented out and only introduced in the FirebaseService Class. But mostly because I didn't find a way to make it work as a abstract class/implementation pair.

My reasoning to have those in the PercistencyService Class was that the frontend currently assumes async communication with the persistency layer - specifically in the action design. I would then assume that we need some sort of success messages flowing back to the frontend.